yt/yt/tests/integration/master/test_users.py

```diff
@@ -818,6 +822,148 @@ def test_password_temporary(self):
+            # Very conservative, this ratio was closer to 100 in local runs.
+            "throttled_to_non_throttled_ratio_threshold": 5,
+            # The throttled execution time should be close to 10 seconds, but let's be generous.
+            "throttled_batch_execution_time_lower_bound": 5.0,
+        }
+
+    @staticmethod
+    def _prepare_read_request_throttling_parameters(user):
+        cypress_limit_path = f"//sys/users/{user}/@read_request_rate_limit"
+        old_limit = get(cypress_limit_path)
+
+        create("map_node", "//tmp/read_me")
+
+        return {
+            "cypress_limit_path": cypress_limit_path,
+            "throttling_limit": 1,
+            "throttling_limit_reset_function": lambda: set(cypress_limit_path, old_limit),
+            "make_batch_request_function": lambda _: make_batch_request("get", path=f"//tmp/read_me/@owner"),
+            "request_count": 10,
+            # Very conservative, this ratio was closer to 100 (or more) in local runs.
+            "throttled_to_non_throttled_ratio_threshold": 5,
+            # The throttled execution time should be close to 10 seconds, but let's be generous.
+            "throttled_batch_execution_time_lower_bound": 5.0,
+        }
+
+    @staticmethod
+    def _prepare_automaton_request_throttling_parameters(_):
+        cypress_limit_path = "//sys/@config/object_service/automaton_request_throttler/limit"
+
+        return {
+            "cypress_limit_path": cypress_limit_path,
+            "throttling_limit": 1,
+            "throttling_limit_reset_function": lambda: remove(cypress_limit_path),
+            "make_batch_request_function": lambda i: make_batch_request("create", type="map_node", path=f"//tmp/alice-{i:02d}"),
+            "request_count": 10,
+            # Very conservative, this ratio was closer to 100 in local runs.
+            "throttled_to_non_throttled_ratio_threshold": 5,
+            # The throttled execution time should be close to 10 seconds, but let's be generous.
+            "throttled_batch_execution_time_lower_bound": 5.0,
+        }
+
+    @authors("achulkov2")
+    @pytest.mark.parametrize("prepare_test_parameters", [
+        _prepare_write_request_throttling_parameters,
+        _prepare_read_request_throttling_parameters,
+        _prepare_automaton_request_throttling_parameters,
+    ])
+    def test_request_throttling(self, prepare_test_parameters):
+        user = "alice"
+        create_user(user)
+
+        params = prepare_test_parameters(user)
+
+        start_time = time.time()
+
+        request_results = execute_batch([params["make_batch_request_function"](i) for i in range(params["request_count"])], authenticated_user=user)
+        for request_result in request_results:
+            raise_batch_error(request_result)
+
+        no_throttling_batch_execution_time = time.time() - start_time
+
+        throttled_batch_execution_time = 0
+
+        with deferred(params["throttling_limit_reset_function"]):
```

Comment:
Shouldn't teardown code handle this? It definitely resets the dynconfig. I'm not so sure about per-user throttlers but I suspect all the users should be removed at teardown anyway.

Reply:
It definitely is. 
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/tests/library/yt_env_setup.py#L1796

Reply:
Removed unnecessary logic, everything seems to work fine.